packages/components/Button/src/Button.tsx

```diff
@@ -34,19 +34,56 @@ export const buttonLookup = (layer: string, state: IPressableState, userProps: B
   );
 };
 
+export const extractOuterStylePropsAndroid = memoize((style) => {
```

Comment:
I think it'd be nice for this to be in its own file which has a forked impl for Android vs. other platforms, so that this Android-only logic only gets bundled into bundles that run Android.

Reply:
Hoping for this function to be used for Shadows as well, do you think it can be placed somewhere more appropriate

Reply:
Shadow has its own version of this (i.e. logic splitting inner/outer props) - it's not exactly a function the way you have it, it's kind of just built into the component: https://github.com/microsoft/fluentui-react-native/blob/main/packages/experimental/Shadow/src/Shadow.tsx.

Saad has brought up that we should find a way to consolidate this logic (I think he might have mentioned this somewhere else in this PR) but his suggestion was to make a rootWithShadow component, or as I understand it just a general component that has an inner and outer view that can be customized, and have Shadow and ripple both use that. I can take on this work but not sure when I'll be able to get to it.

Reply:
Based on how things are currently, the structure for the Button on Android is currently -
`<View><Pressable>{Inner Stuff}</Pressable></View>` 
While for Button on iOS for the shadow, it is -
`<View><View>{Inner Stuff}</View></View>` so it would not be possible to do this rn.

But! This can be done once @Saadnajmi 's PR of moving to [usePressableState](https://github.com/microsoft/fluentui-react-native/pull/2153) goes through (It will enable us to use `<View>` directly instead of `<Pressable>` to provide the ripple effect)